test/src/unit-regression2.cpp

```diff
@@ -793,8 +793,8 @@ TEST_CASE("regression tests 2")
         const auto j_path = j.get<nlohmann::detail::std_fs::path>();
         CHECK(j_path == text_path);
 
-#ifndef _MSC_VER
-        // works everywhere but on MSVC
+#if !defined(_MSC_VER) && !(defined(__GNUC__) && __GNUC__ == 8 && __GNUC_MINOR__ < 4)
+        // works everywhere but on MSVC and GCC <8.4
```

Comment:
`<filesystem>` is only available since GCC 8 so this check should be enough. I tested 8.1, 8.2, and 8.3 on Compiler explorer.